src/NServiceBus.MessagingBridge/EndpointRegistry.cs

```diff
@@ -60,16 +63,18 @@ public TargetEndpointDispatcher GetTargetEndpointDispatcher(string sourceEndpoin
         throw new Exception($"No target endpoint dispatcher could be found for endpoint: {sourceEndpointName}. Ensure names have correct casing as mappings are case-sensitive. Nearest configured match: {nearestMatch}");
     }
 
-    public string TranslateToTargetAddress(string sourceAddress)
+    public bool TryTranslateToTargetAddress(string sourceAddress, out (string targetAddress, string nearestMatch) result)
     {
-        if (targetEndpointAddressMappings.TryGetValue(sourceAddress, out var targetAddress))
+        if (targetEndpointAddressMappings.TryGetValue(sourceAddress, out result.targetAddress))
         {
-            return targetAddress;
+            result.nearestMatch = result.targetAddress;
+            return true;
         }
 
-        var nearestMatch = GetClosestMatchForExceptionMessage(sourceAddress, targetEndpointAddressMappings.Keys);
+        logger.LogWarning($"Could not translate {sourceAddress} address. Consider using `.HasEndpoint()` method to add missing endpoint declaration.");
```

Comment:
I'm not sure if warning should be logged here - won't it lead to duplicate logging of the error in cases where an exception is ultimately thrown?
Can `bestEffort` be used to switch between logging/throwing in `TransformAddressHeader`?